File: javaee/jaxrs/src/main/java/cz/ladicek/swarm/tiny/javaeeJaxrs/HelloResource.java
package cz.ladicek.swarm.tiny.javaeeJaxrs;

import javax.ws.rs.GET;
import javax.ws.rs.Path;
import javax.ws.rs.core.Response;

@Path("/")
@javax.inject.Singleton
public class HelloResource {
    @GET
    public String get() {
        return "Hello from JAX-RS";
    }
}

File: kotlin/src/main/kotlin/cz/ladicek/swarm/tiny/kotlin/HelloResource.kt
package cz.ladicek.swarm.tiny.kotlin

import io.swagger.annotations.Api
import io.swagger.annotations.SwaggerDefinition
import io.swagger.annotations.Tag
import javax.ws.rs.DefaultValue
import javax.ws.rs.GET
import javax.ws.rs.Path
import javax.ws.rs.QueryParam

@Path("/")
@Api
@SwaggerDefinition(tags = [Tag(name = "Hello", description = "Hello World service")])
class HelloResource {
    @GET
    fun get(@QueryParam("name") @DefaultValue("World") name: String): String = "Hello, $name!"
}
